server-spi-private/src/main/java/org/keycloak/tracing/TracingProvider.java

```diff
@@ -170,7 +170,20 @@ default Span startSpan(Class<?> tracerClass, String spanSuffix) {
      * @param execution   code that is executed and traced
      */
     default void trace(Class<?> tracerClass, String spanSuffix, Consumer<Span> execution) {
-        trace(tracerClass.getName(), tracerClass.getSimpleName() + "." + spanSuffix, execution);
+        String className = prepareClassName(tracerClass);
+        trace(tracerClass.getName(), className + "." + spanSuffix, execution);
+    }
+
+    private static String prepareClassName(Class<?> tracerClass) {
```

Comment:
just a consideration - If we'd have the dependency on `org.apache.commons:commons-lang3` in `server-spi-private`, we could use `StringUtils.substringAfterLast()`. We already have it in quarkus module. It'd be good to not have such "util" method in the interface. But it's only a nitpick.

Reply:
@ahus1 

> Some people might depend on this module even though it is private, so I'd rather not have dependencies there.

+1

> We could remove default method and have all implementations implement it, and then we could have a dependency there. WDYT?

Nah, it's not a big deal, we can leave it as is. Some improvement would be to add the util method to our KC `StringUtil` class, where it belongs. But I'll leave the decision up to you - maybe it might be useful even in other cases in our codebase.

Reply:
I would like to leave it as is for now. Please then approve the PR, and I'll ask a maintainer to merge it.

Reply:
Ok, agree.